src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -123,29 +122,7 @@ internal partial class Binder
 
         private bool IsCheckedConversion(TypeSymbol source, TypeSymbol target)
         {
-            Debug.Assert((object)target != null);
-
-            if ((object)source == null || !CheckOverflowAtRuntime)
-            {
-                return false;
-            }
-
-            if (source.IsDynamic())
-            {
-                return true;
-            }
-
-            SpecialType sourceST = source.StrippedType().EnumUnderlyingType().SpecialType;
-            SpecialType targetST = target.StrippedType().EnumUnderlyingType().SpecialType;
-
-            // integral to double or float is never checked, but float/double to integral 
-            // may be checked.
-            bool sourceIsNumeric = SpecialType.System_Char <= sourceST && sourceST <= SpecialType.System_Double;
-            bool targetIsNumeric = SpecialType.System_Char <= targetST && targetST <= SpecialType.System_UInt64;
-
-            return
-                sourceIsNumeric && (targetIsNumeric || target.IsPointerType()) ||
-                targetIsNumeric && source.IsPointerType();
```

Comment:
This method calls `NeedsCheckedConversion(..., compareNumericTypes: false)` to skip that last check, for compatibility with the previous implementation.

That said, I'd prefer to remove the checks and just use `CheckOverflowAtRuntime` as @AlekseyTs suggested. I'll test that.